Game/Scripts/DebugGame.cpp

```diff
@@ -0,0 +1,237 @@
+{
+	REGISTER_FIELD(setPlayer, GameObject*);
+	REGISTER_FIELD(debugPoint1, GameObject*);
+	REGISTER_FIELD(debugPoint2, GameObject*);
+	REGISTER_FIELD(debugPoint3, GameObject*);
+	REGISTER_FIELD(debugPoint4, GameObject*);
+	REGISTER_FIELD(debugPoint5, GameObject*);
+	//REGISTER_FIELD_WITH_ACCESSORS(debugCurrentPos, int);
+
+}
+
+void DebugGame::Start()
+{
+	//ImmortalityStart
+	player = setPlayer->GetComponent<ComponentPlayer>();
+
+	//GameManager* manager = GameManager::GetInstance();
+
+	/*std::vector<ComponentScript*> gameObjectScripts =*/
+		/*setPlayer->GetComponentsByType<ComponentScript>(ComponentType::SCRIPT);*/
+
+	playerHealthSystem = setPlayer->GetComponent<HealthSystem>();
+	playerAttackScript = setPlayer->GetComponent<BixAttackScript>();
+	playerMoveScript = setPlayer->GetComponent<PlayerMoveScript>();
+	playerJumpScript = setPlayer->GetComponent<PlayerJumpScript>();
+	playerRotationScript = setPlayer->GetComponent<PlayerRotationScript>();
+
+	isDebugModeActive = false;
+
+	//TeleportStart
+
+	if (debugPoint1)
+	{
+		debugPoints.push_back(debugPoint1->GetComponent<ComponentTransform>());
+	}
+
+	if (debugPoint2)
+	{
+		debugPoints.push_back(debugPoint2->GetComponent<ComponentTransform>());
+	}
+
+	if (debugPoint3)
+	{
+		debugPoints.push_back(debugPoint3->GetComponent<ComponentTransform>());
+	}
+
+	if (debugPoint4)
+	{
+		debugPoints.push_back(debugPoint4->GetComponent<ComponentTransform>());
+	}
+
+	if (debugPoint5)
+	{
+		debugPoints.push_back(debugPoint5->GetComponent<ComponentTransform>());
+	}
+
+	playerRigidBody = setPlayer->GetComponent<ComponentRigidBody>();
+	playerTransform = setPlayer->GetComponent<ComponentTransform>();
+
+	currentdDebugPointTransform = debugPoints.front();
+
+	debugCurrentPosIndex = 0;
+	playerOnLocation = false;
+}
+
+void DebugGame::Update(float deltaTime)
+{
+	ModuleInput* input = App->GetModule<ModuleInput>();
+	
+	//INPUTS
+
+	if (input->GetKey(SDL_SCANCODE_B) == KeyState::DOWN)
+	{
+		ActivateDebugMode();
+	}
+
+	if (input->GetKey(SDL_SCANCODE_4) == KeyState::DOWN && isDebugModeActive)
+	{
+		GodCamera();
+	}
+
+	if (input->GetKey(SDL_SCANCODE_7) == KeyState::DOWN && isDebugModeActive)
+	{
+		FillHealth();
+	}
+
+	if (input->GetKey(SDL_SCANCODE_8) == KeyState::DOWN && isDebugModeActive)
+	{
+
+		BeImmortal();
+	}
+
+	if (input->GetKey(SDL_SCANCODE_9) == KeyState::DOWN && isDebugModeActive)
+	{
+		
+		DeathTouch();
+	}
+
+	if (input->GetKey(SDL_SCANCODE_0) == KeyState::DOWN && isDebugModeActive)
+	{
+		Teleport();
+		playerOnLocation = false;
+		
+	}
+
+	//TELEPORTMOV
+	if(!playerOnLocation && isDebugModeActive)
+	{
+		for(const ComponentTransform* debugPointTransform : debugPoints)
+		{
+			if (playerTransform->GetGlobalPosition().Equals(debugPointTransform->GetGlobalPosition(), 1.0f))
+			{
+				playerRigidBody->SetIsTrigger(false);
+				playerRigidBody->DisablePositionController();
+				playerOnLocation = true;
+				break;
+			}
+		}
+	}
+	
+
+
+}
+
+void DebugGame::ActivateDebugMode()
```

Comment:
Here we not only activate debugMode we also deactivate it, i suggest
```suggestion
void DebugGame::SwitchDebugMode()
```